ql/src/java/org/apache/hadoop/hive/ql/ppd/OpProcFactory.java

```diff
@@ -728,6 +763,166 @@ private void applyFilterTransitivity(JoinOperator join, int targetPos, OpWalkerI
         }
       }
     }
+
+    private Set<ExprNodeColumnDesc> collectColumnsInPredicates(List<ExprNodeDesc> predicates) {
```

Comment:
Can we add a few high level comments to these new private methods describing what they do?

Reply:
added javadoc